src/ftp/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+use self::retrieve::Retrieve;
+use std::fmt::Display;
+use std::io::{BufRead, BufReader, Write};
+use std::net::TcpStream;
+use thiserror::Error;
+
+pub mod retrieve;
+
+/// The reason we need to implement our own FTP client is because all of other FTP client is not
+/// support custom command.
```

Comment:
slight grammer issue here. Should be "The reason we need to implement our own FTP client is because no other FTP client supports the ```custom``` command".